source/Symbol/SwiftASTContext.cpp

```diff
@@ -3128,6 +3131,81 @@ class StoringDiagnosticConsumer : public swift::DiagnosticConsumer {
   unsigned m_num_errors = 0;
   bool m_colorize;
 };
+
+/// Implements a swift::DWARFImporterDelegate to look up Clang types in DWARF.
+///
+/// During compile time, ClangImporter-imported Clang modules are compiled with
+/// -gmodules, which emits a DWARF rendition of all types defined in the module
+/// into the .pcm file. On Darwin, these types can be collected by
+/// dsymutil. This delegate allows DWARFImporter to ask LLDB to look up a Clang
+/// type by name, synthesize a Clang AST from it. DWARFImporter then hands this
+/// Clang AST to ClangImporter to import the type into Swift.
+class SwiftDWARFImporterDelegate : public swift::DWARFImporterDelegate {
+  SwiftASTContext &m_swift_ast_ctx;
+public:
+  SwiftDWARFImporterDelegate(SwiftASTContext &swift_ast_ctx)
+      : m_swift_ast_ctx(swift_ast_ctx) {}
+
+  void lookupValue(StringRef name,
+                   llvm::Optional<swift::Demangle::Node::Kind> kind,
+                   llvm::SmallVectorImpl<clang::Decl *> &results) override {
+    std::vector<CompilerContext> decl_context;
+    ConstString name_cs(name);
+    decl_context.push_back({CompilerContextKind::Structure, name_cs});
+    auto *dwarf_importer = m_swift_ast_ctx.GetDWARFImporter();
+    if (!dwarf_importer)
+      return;
+    Module *module = m_swift_ast_ctx.GetModule();
+    if (!module)
+      return;
+
+    TypeList clang_types;
+    const bool exact_match = true;
+    const uint32_t max_matches = UINT32_MAX;
```

Comment:
`std::numeric_limits<int>::max()`

Reply:
Looking at the rest of the file UINT32_MAX seems to be more idiomatic. Of course the API should really take an optional instead...